lib/protocol/protocol.js

```diff
@@ -423,8 +427,13 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       }
 
       httpResBody = formatStatus(httpResBody, httpStatus, currentProtocol);
-
       res.status(httpStatus).json(httpResBody);
+
+      // in certain situations the socket should not be kept alive
+      if (terminateSocket) {
+        SESSIONS_CACHE.getLogger(req.params.sessionId || newSessionId, currentProtocol).debug(`Destroying socket connection`);
+        res.socket.destroy();
```

Comment:
should we rather invoke `end` ?

Reply:
`res.end()` simply signals that the response is fully written. Also, the previous call to `res.json()` causes the `end` to get called.

The issue here is that the socket remains, if it is keep-alive.